src/vms/pvm/etna-builder/builder.ts

```diff
@@ -0,0 +1,1258 @@
+import {
+  Bytes,
+  Id,
+  Int,
+  TransferableInput,
+  TransferableOutput,
+} from '../../../serializable';
+import { BaseTx as AvaxBaseTx } from '../../../serializable/avax';
+import type { Utxo } from '../../../serializable/avax/utxo';
+import {
+  AddPermissionlessDelegatorTx,
+  AddPermissionlessValidatorTx,
+  AddSubnetValidatorTx,
+  BaseTx,
+  CreateChainTx,
+  CreateSubnetTx,
+  ExportTx,
+  ImportTx,
+  RemoveSubnetValidatorTx,
+  SubnetValidator,
+  TransferSubnetOwnershipTx,
+} from '../../../serializable/pvm';
+import { createSignerOrSignerEmptyFromStrings } from '../../../serializable/pvm/signer';
+import { AddressMaps, addressesFromBytes, isTransferOut } from '../../../utils';
+import { matchOwners } from '../../../utils/matchOwners';
+import { compareTransferableOutputs } from '../../../utils/sort';
+import { baseTxUnsafePvm, type SpendOptions, UnsignedTx } from '../../common';
+import { defaultSpendOptions } from '../../common/defaultSpendOptions';
+import type { Dimensions } from '../../common/fees/dimensions';
+import { addDimensions, createDimensions } from '../../common/fees/dimensions';
+import type { Context } from '../../context';
+import {
+  ID_LEN,
+  INTRINSIC_ADD_PERMISSIONLESS_DELEGATOR_TX_COMPLEXITIES,
+  INTRINSIC_ADD_PERMISSIONLESS_VALIDATOR_TX_COMPLEXITIES,
+  INTRINSIC_ADD_SUBNET_VALIDATOR_TX_COMPLEXITIES,
+  INTRINSIC_BASE_TX_COMPLEXITIES,
+  INTRINSIC_CREATE_CHAIN_TX_COMPLEXITIES,
+  INTRINSIC_CREATE_SUBNET_TX_COMPLEXITIES,
+  INTRINSIC_EXPORT_TX_COMPLEXITIES,
+  INTRINSIC_IMPORT_TX_COMPLEXITIES,
+  INTRINSIC_REMOVE_SUBNET_VALIDATOR_TX_COMPLEXITIES,
+  INTRINSIC_TRANSFER_SUBNET_OWNERSHIP_TX_COMPLEXITIES,
+  getAuthComplexity,
+  getInputComplexity,
+  getOutputComplexity,
+  getOwnerComplexity,
+  getSignerComplexity,
+} from '../txs/fee';
+import { spend } from './spend';
+
+const getAddressMaps = ({
+  inputs,
+  inputUTXOs,
+  minIssuanceTime,
+  fromAddressesBytes,
+}: {
+  inputs: readonly TransferableInput[];
+  inputUTXOs: readonly Utxo[];
+  minIssuanceTime: bigint;
+  fromAddressesBytes: readonly Uint8Array[];
+}): AddressMaps => {
+  return AddressMaps.fromTransferableInputs(
+    inputs,
+    inputUTXOs,
+    minIssuanceTime,
+    fromAddressesBytes,
+  );
+};
+
+const getMemoComplexity = (
+  spendOptions: Required<SpendOptions>,
+): Dimensions => {
+  return createDimensions(spendOptions.memo.length, 0, 0, 0);
+};
+
+/**
+ * Common properties used in all PVM transaction builder functions.
+ */
+type CommonTxProps = Readonly<{
+  /**
+   * List of addresses that are used for selecting which UTXOs are signable.
+   */
+  fromAddressesBytes: readonly Uint8Array[];
+  options?: SpendOptions;
+  /**
+   * List of UTXOs that are available to be spent.
+   */
+  utxos: readonly Utxo[];
+}>;
+
+type TxProps<T extends Record<string, unknown>> = CommonTxProps & Readonly<T>;
+
+type TxBuilderFn<T extends TxProps<Record<string, unknown>>> = (
+  props: T,
+  context: Context,
```

Comment:
I'm going to leave it as is for now, but I suspect this may change on the next revision of changes once the APIs are in place for getting the gasprice and weights.

Once I know how that looks we can make a better determination on if these values should be in the context or not.